misc/sync-typeshed.py

```diff
@@ -148,58 +149,66 @@ def main() -> None:
+        shutil.copytree(tmp_patches, typeshed_patches)
+        if subprocess.run(["git", "diff", "--quiet", "--exit-code"], check=False).returncode != 0:
+            subprocess.run(["git", "commit", "-am", "Update typeshed patches"], check=True)
+
+        if not args.typeshed_dir:
+            tmp_typeshed = os.path.join(tmpdir, "typeshed")
+            os.makedirs(tmp_typeshed)
+            # Clone typeshed repo if no directory given.
+            print(f"Cloning typeshed in {tmp_typeshed}...")
             subprocess.run(
-                ["git", "clone", "https://github.com/python/typeshed.git"], check=True, cwd=tempdir
+                ["git", "clone", "https://github.com/python/typeshed.git"],
+                check=True,
+                cwd=tmp_typeshed,
             )
-            repo = os.path.join(tempdir, "typeshed")
+            repo = os.path.join(tmp_typeshed, "typeshed")
             commit = update_typeshed(repo, args.commit)
-    else:
-        commit = update_typeshed(args.typeshed_dir, args.commit)
+        else:
+            commit = update_typeshed(args.typeshed_dir, args.commit)
 
-    assert commit
+        assert commit
 
-    # Create a commit
-    message = textwrap.dedent(
-        f"""\
-        Sync typeshed
+        # Create a commit
+        message = textwrap.dedent(
+            f"""\
+            Sync typeshed
 
-        Source commit:
-        https://github.com/python/typeshed/commit/{commit}
-        """
-    )
-    subprocess.run(["git", "add", "--all", os.path.join("mypy", "typeshed")], check=True)
-    subprocess.run(["git", "commit", "-m", message], check=True)
-    print("Created typeshed sync commit.")
-
-    commits_to_cherry_pick = [
-        "5c00e362d",  # LiteralString reverts
-        "44bc98bd5",  # sum reverts
-        "61a490091",  # ctypes reverts
-    ]
-    for commit in commits_to_cherry_pick:
-        try:
-            subprocess.run(["git", "cherry-pick", commit], check=True)
-        except subprocess.CalledProcessError:
-            if not sys.__stdin__.isatty():
-                # We're in an automated context
-                raise
-
-            # Allow the option to merge manually
-            print(
-                f"Commit {commit} failed to cherry pick."
-                " In a separate shell, please manually merge and continue cherry pick."
-            )
-            rsp = input("Did you finish the cherry pick? [y/N]: ")
-            if rsp.lower() not in {"y", "yes"}:
-                raise
-        print(f"Cherry-picked {commit}.")
+            Source commit:
+            https://github.com/python/typeshed/commit/{commit}
+            """
+        )
+        subprocess.run(["git", "add", "--all", os.path.join("mypy", "typeshed")], check=True)
+        subprocess.run(["git", "commit", "-m", message], check=True)
+        print("Created typeshed sync commit.")
+
+        patches = glob.glob(os.path.join(typeshed_patches, "*.patch"))
```

Comment:
Sort the patches to make sure they are applied in a predictable order on all file systems?